src/parser.rs

```diff
@@ -125,6 +141,32 @@ pub fn inline<T>(result: Result<T, ParserError>) -> Result<T, (ParserError, Toke
 	result.map_err(|error| (error, Token::Terminator))
 }
 
+pub fn animation(lexer: &mut Lexer, check_with: bool) -> Result<Option<AnimationDeclaration>, (ParserError, Token)> {
+	if check_with {
+		match inline(lexer.token())? {
+			None | Some(Token::Terminator) => return Ok(None),
+			Some(Token::Identifier(string)) if string == "with" => {}
+			Some(token) => return Err((ParserError::UnexpectedToken, token)),
+		}
+	}
+	let name = inline(lexer.identifier())?;
+	inline(lexer.expect(Token::SquareOpen))?;
+	let mut arguments = Vec::new();
+	while let Some(token) = inline(lexer.token())? {
+		if token == Token::SquareClose { break; }
+		if !arguments.is_empty() {
+			inline(lexer.expect(Token::ListSeparator))?
+		}
+		let arg = match token {
+			Token::Underscore => None,
+			Token::Numeric(n) => Some(n),
```

Comment:
Use `numeric` or `number` instead of just `n`. Avoid single letter variable names.

Reply:
Done